src/sagemaker_sklearn_extension/preprocessing/encoders.py

```diff
@@ -423,3 +423,168 @@ def inverse_transform(self, y):
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to determine the categories of each feature, assuming the input parameter categories equals 'auto'
+
+        Returns
+        -------
+        self
+
+        """
+        # sklearn.preprocessing._BaseEncoder uses _categories due to deprecations in other classes
+        # can be removed once deprecations are removed
+        self._categories = self.categories
+        self._fit(X, handle_unknown="unknown")
+        return self
+
+    def transform(self, X):
+        """Transform X to ordinal integers.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to encode.
+
+        Returns
+        -------
+        X_out : sparse matrix or a 2-d array
+            Transformed input.
+
+        """
+        X_int, X_mask = self._transform(X, handle_unknown="unknown")
+        # assign the unknowns an integer indicating they are unknown. The largest integer is always reserved for
+        # unknowns
+        for col in range(X_int.shape[1]):
+            mask = X_mask[:, col]
+            X_int[~mask, col] = self.categories_[col].shape[0]
+
+        return X_int.astype(self.dtype, copy=False)
+
+    def inverse_transform(self, X):
+        """Convert the data back to the original representation.
+        In slots where the encoding is that of an unrecognised category, the output of the inverse transform is np.nan
+        for float or complex arrays, and None otherwise
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape [n_samples, n_encoded_features]
+            The transformed data.
+
+        Returns
+        -------
+        X_tr : array-like, shape [n_samples, n_features]
+            Inverse transformed array.
+
+        Notes
+        -----
+        Most of the logic is copied from sklearn.preprocessing.OrdinalEncoder.inverse_transform. The difference is in
+        handling unknown values.
+
+        """
+        check_is_fitted(self, "categories_")
+        X = check_array(X, accept_sparse="csr", dtype="numeric")
+
+        n_samples, _ = X.shape
+        n_features = len(self.categories_)
+
+        # validate shape of passed X
+        msg = "Shape of the passed X data is not correct. Expected {0} " "columns, got {1}."
+        if X.shape[1] != n_features:
+            raise ValueError(msg.format(n_features, X.shape[1]))
+
+        # create resulting array of appropriate dtype
+        dt = np.find_common_type([cat.dtype for cat in self.categories_], [])
+        X_tr = np.empty((n_samples, n_features), dtype=dt)
+
+        found_unknown = {}
+        for i in range(n_features):
+            labels = X[:, i].astype("int64", copy=False)
+            known_mask = labels != self.categories_[i].shape[0]
+            labels *= known_mask
+            X_tr[:, i] = self.categories_[i][labels]
```

Comment:
When I tested out `inverse_transform` with an unknown value, I get an error. See console output below

```
>>> enc.transform([['Female', 3], ['Male', 1], ['Fmale', 4]])
array([[0., 2.],
       [1., 0.],
       [2., 3.]], dtype=float32)
>>> enc.inverse_transform([[1, 0], [0, 1]])
array([['Male', 1],
       ['Female', 2]], dtype=object)
>>> X = [['Male', 1], ['Female', 3], ['Female', 2]]
>>> enc.fit(X)
RobustOrdinalEncoder(categories='auto', dtype=<class 'numpy.float32'>)
>>> enc.transform([['Female', 3], ['Male', 1]])
array([[0., 2.],
       [1., 0.]], dtype=float32)
>>> enc.categories_
[array(['Female', 'Male'], dtype=object), array([1, 2, 3], dtype=object)]
>>> enc.transform([['Female', 3], ['Male', 1], ['?', 4]])
array([[0., 2.],
       [1., 0.],
       [2., 3.]], dtype=float32)
>>> enc.inverse_transform([[1, 0], [0, 1]])
array([['Male', 1],
       ['Female', 2]], dtype=object)
>>> enc.inverse_transform([[1, 0], [0, 1], [9, 9]])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/wuwilton/miniconda3/lib/python3.7/site-packages/sagemaker_sklearn_extension/preprocessing/encoders.py", line 578, in inverse_transform
    X_tr[:, i] = self.categories_[i][labels]
IndexError: index 9 is out of bounds for axis 0 with size 2
```

Reply:
This should be an error. inverse_transform doesn't have to deal with any integer. It needs to know that a specific integer (in this case, 2 for column 0 and 3 for column 1) encode an unknown category